extensions/kubernetes-client/deployment/src/main/java/io/quarkus/kubernetes/client/deployment/KubernetesClientProcessor.java

```diff
@@ -0,0 +1,55 @@
+package io.quarkus.kubernetes.client.deployment;
+
+import javax.inject.Inject;
+
+import org.jboss.jandex.DotName;
+
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
+import io.quarkus.deployment.builditem.ExtensionSslNativeSupportBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.substrate.ReflectiveClassBuildItem;
+
+public class KubernetesClientProcessor {
+
+    @Inject
+    BuildProducer<FeatureBuildItem> featureProducer;
+
+    @Inject
+    BuildProducer<ReflectiveClassBuildItem> reflectiveClasses;
+
+    @BuildStep
+    public void process(CombinedIndexBuildItem combinedIndexBuildItem,
+            BuildProducer<ExtensionSslNativeSupportBuildItem> sslNativeSupport) {
+        featureProducer.produce(new FeatureBuildItem(FeatureBuildItem.KUBERNETES_CLIENT));
+
+        final String[] modelClasses = combinedIndexBuildItem.getIndex()
+                .getAllKnownImplementors(DotName.createSimple("io.fabric8.kubernetes.api.model.KubernetesResource"))
+                .stream()
+                .map(c -> c.name().toString())
+                .toArray(String[]::new);
+        reflectiveClasses.produce(ReflectiveClassBuildItem.weakClass(modelClasses));
+
+        final String[] doneables = combinedIndexBuildItem.getIndex()
+                .getAllKnownImplementors(DotName.createSimple("io.fabric8.kubernetes.api.model.Doneable"))
+                .stream()
+                .map(c -> c.name().toString())
+                .toArray(String[]::new);
+        reflectiveClasses.produce(ReflectiveClassBuildItem.weakClass(doneables));
+
+        final String[] deserializerClasses = combinedIndexBuildItem.getIndex()
+                .getAllKnownSubclasses(DotName.createSimple("com.fasterxml.jackson.databind.JsonDeserializer"))
+                .stream()
+                .map(c -> c.name().toString())
+                .filter(s -> s.startsWith("io.fabric8.kubernetes"))
```

Comment:
Wondering if this filter is a good idea? We don't expect the user to add custom serializers for their model?

Reply:
Most likely a user won't add a custom deserializer. What I wanted to do here by adding this filter is avoid a situation were adding the Kubernetes Client dependency suddently makes a host of other unrelated serializers available for reflection.
I don't mind dropping the filter, but I would say that perhas we should keep as a first approach and see if there are actually any problems caused by it.

Reply:
I think at some point, it might make sense to have a Jackson extension to deal with that but for now, I suppose it's good enough.

Reply:
Yeah, I could see a Jackson extension making sense